peer/single.go

```
@@ -61,12 +71,23 @@ func (s *Single) NotifyStatusChanged(_ peer.Identifier) {
 
 // Start is a noop
 func (s *Single) Start() error {
+	return s.once.Start(s.start)
+}
+
+func (s *Single) start() error {
+	p, err := s.t.RetainPeer(s.pid, s)
+	s.p = p
+	s.err = err
 	return nil
```

shouldn't we return the error here?